sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -48,20 +49,22 @@
     private final GatewayServiceConfigurationReader serviceConfigReader;
     final Map<URI, EndpointCache> addressCacheByEndpoint;
     private final boolean tcpConnectionEndpointRediscoveryEnabled;
+    private ApiType apiType;
 
     private HttpClient httpClient;
 
     public GlobalAddressResolver(
-            DiagnosticsClientContext diagnosticsClientContext,
-            HttpClient httpClient,
-            GlobalEndpointManager endpointManager,
-            Protocol protocol,
-            IAuthorizationTokenProvider tokenProvider,
-            RxCollectionCache collectionCache,
-            RxPartitionKeyRangeCache routingMapProvider,
-            UserAgentContainer userAgentContainer,
-            GatewayServiceConfigurationReader serviceConfigReader,
-            ConnectionPolicy connectionPolicy) {
+        DiagnosticsClientContext diagnosticsClientContext,
+        HttpClient httpClient,
+        GlobalEndpointManager endpointManager,
+        Protocol protocol,
+        IAuthorizationTokenProvider tokenProvider,
+        RxCollectionCache collectionCache,
+        RxPartitionKeyRangeCache routingMapProvider,
+        UserAgentContainer userAgentContainer,
+        GatewayServiceConfigurationReader serviceConfigReader,
+        ConnectionPolicy connectionPolicy,
+        ApiType apiType) {
```

Comment:
this is a cascading change. we should avoid adding `apiType` everywhere. otherwise we will have to add this everywhere in every single cache we have.

instead we should have `apiType` in the place where you are sending the request out. meaning you can add apiType in the GatewayStoreModel to ensure any request sent out will have the proper header.